NullPointerException if tomcat-users.xml doesn't have a role associated for user

If tomcat-users.xml doesn't have a role associated with a user (as an attribute), it's resulting in NullPointerException. That in turn prevents the user to login to hawtio UI. 

Following is an example content for the tomcat-users.xml file which is resulting in NullPointerException: 
```<user username="admin" password="password"/>```

Exception Stack trace:
```
javax.security.auth.login.LoginException: java.lang.NullPointerException
	at io.hawt.web.tomcat.TomcatUserDatabaseLoginContext.login(TomcatUserDatabaseLoginContext.java:157)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at io.hawt.system.Authenticator.doAuthenticate(Authenticator.java:131)
	at io.hawt.system.Authenticator.authenticate(Authenticator.java:92)
	at io.hawt.web.AuthenticationFilter.doFilter(AuthenticationFilter.java:163)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at io.hawt.web.CORSFilter.doFilter(CORSFilter.java:42)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at io.hawt.web.XFrameOptionsFilter.doFilter(XFrameOptionsFilter.java:28)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at io.hawt.web.SessionExpiryFilter.process(SessionExpiryFilter.java:117)
	at io.hawt.web.SessionExpiryFilter.doFilter(SessionExpiryFilter.java:57)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:423)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1079)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:620)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1747)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1706)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	at io.hawt.web.tomcat.TomcatUserDatabaseLoginContext.getUserPasswordRole(TomcatUserDatabaseLoginContext.java:197)
	at io.hawt.web.tomcat.TomcatUserDatabaseLoginContext.login(TomcatUserDatabaseLoginContext.java:135)
	... 41 more
```

TomcatUserDatabaseLoginContext.getUserPasswordRole(String username) assumes that there is always role associated with the user:
```
String nRoles = node.getAttributes().getNamedItem("roles").getNodeValue();
```
If accepted as an issue, I can submit a pull request.

--- a/hawtio-system/src/main/java/io/hawt/web/tomcat/TomcatUserDatabaseLoginContext.java
+++ b/hawtio-system/src/main/java/io/hawt/web/tomcat/TomcatUserDatabaseLoginContext.java
@@ -5,6 +5,7 @@ import org.apache.commons.codec.digest.DigestUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
 import org.w3c.dom.Node;
 import org.w3c.dom.NodeList;
 
@@ -138,10 +139,12 @@ public class TomcatUserDatabaseLoginContext implements LoginModule {
                     LOG.trace("Login denied due password did not match");
                     return false;
                 }
-                String[] roles = user[2].split(",");
-                for (String role : roles) {
-                    LOG.trace("User {} has role {}", username, role);
-                    subject.getPrincipals().add(new TomcatPrincipal(role));
+                if (user[2] != null) {
+                    String[] roles = user[2].split(",");
+                    for (String role : roles) {
+                        LOG.trace("User {} has role {}", username, role);
+                        subject.getPrincipals().add(new TomcatPrincipal(role));
+                    }
                 }
             } else {
                 LOG.trace("Login denied due user not found");
@@ -193,7 +196,9 @@ public class TomcatUserDatabaseLoginContext implements LoginModule {
             Node node = users.item(i);
             String nUsername = node.getAttributes().getNamedItem("username").getNodeValue();
             String nPassword = node.getAttributes().getNamedItem("password").getNodeValue();
-            String nRoles = node.getAttributes().getNamedItem("roles").getNodeValue();
+            NamedNodeMap attributes = node.getAttributes();
+            Node roleNode = attributes != null ? attributes.getNamedItem("roles") : null;
+            String nRoles = roleNode != null ? roleNode.getNodeValue() : null;
             if (username.equals(nUsername)) {
                 return new String[]{username, nPassword, nRoles};
             }

